src/backend/gpopt/translate/CQueryMutators.cpp

```diff
@@ -13,21 +13,16 @@
 //
 //---------------------------------------------------------------------------
 
-extern "C" {
-#include "postgres.h"
-
-#include "nodes/makefuncs.h"
-#include "nodes/parsenodes.h"
-#include "nodes/plannodes.h"
-#include "optimizer/walkers.h"
-}
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wcast-function-type"
```

Comment:
When a warning is ignored for the entire file, can we not use push-pop here and in other similar places?

Reply:
ok